### PR TITLE
Adding pvlan option for l2 and fixing passing none for no pvlan type

### PR DESCRIPTION
**Pull request by davidjumani**
This adds the option to create PVLAN for L2 networks, as well as fixes the issue caused by passing 'none' as the PVLAN type when no secondary VLAN type is selected

### CODE REVIEW

**Comment by rhtyd** [comment]
@borisstoyanov @vladimirpetrov - please ping me once you confirm testing the feature
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
Does it cover all cases for https://github.com/apache/cloudstack/pull/3732 ? @davidjumani ?

**Comment by davidjumani** [comment]
@rhtyd Yes it does!

**Comment by borisstoyanov** [comment]
@blueorangutan test

File: src/views/network/CreateL2NetworkForm.vue [hunk 1]
```diff
@@ -109,6 +109,32 @@
           <a-form-item :label="$t('label.bypassvlanoverlapcheck')" v-if="!this.isObjectEmpty(this.selectedNetworkOffering) && this.selectedNetworkOffering.specifyvlan">
             <a-switch v-decorator="['bypassvlanoverlapcheck']" />
           </a-form-item>
+          <a-form-item :label="$t('label.isolatedpvlantype')" v-if="!this.isObjectEmpty(this.selectedNetworkOffering) && this.selectedNetworkOffering.specifyvlan">
+            <a-radio-group
+              v-decorator="['isolatedpvlantype', {
+                initialValue: this.isolatePvlanType
+              }]"
+              buttonStyle="solid"
+              @change="selected => { this.handleIsolatedPvlanTypeChange(selected.target.value) }">
```
Comment (shwstppr) [inline src/views/network/CreateL2NetworkForm.vue:118]: minor: can be simplified
```suggestion
              @change="selected => { this.isolatePvlanType = selected.target.value }">
```

File: src/views/network/CreateL2NetworkForm.vue [hunk 2]
```diff
@@ -228,6 +255,9 @@ export default {
       this.selectedZone = zone
       this.updateVPCCheckAndFetchNetworkOfferingData()
     },
+    handleIsolatedPvlanTypeChange (pvlan) {
+      this.isolatePvlanType = pvlan
+    },
```
Comment (shwstppr) [inline src/views/network/CreateL2NetworkForm.vue:260]:
```suggestion
```

File: src/views/network/CreateSharedNetworkForm.vue [hunk 3]
```diff
@@ -589,11 +589,11 @@ export default {
...
         }
-        if (this.isValidValueForKey(values, 'isolatedpvlantype')) {
+        if (this.isValidValueForKey(values, 'isolatedpvlantype') && values.isolatedpvlantype !== 'none') {
           params.isolatedpvlantype = values.isolatedpvlantype
-        }
-        if (this.isValidValueForKey(values, 'isolatedpvlan')) {
-          params.isolatedpvlan = values.isolatedpvlan
+          if (this.isValidValueForKey(values, 'isolatedpvlan')) {
+            params.isolatedpvlan = values.isolatedpvlan
+          }
```
Comment (shwstppr) [inline src/views/network/CreateSharedNetworkForm.vue:596]: Not sure if this really needs to go inside `isolatedpvlantype` if block. Anyway we are not showing `isolatedpvlan` field for `none` type

Comment (davidjumani) [reply src/views/network/CreateSharedNetworkForm.vue:596]: so we're verifying it only if the pvlan type is not none

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@davidjumani can you check and fix label on L2 form? (label.promiscuous)

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

File: src/views/network/CreateL2NetworkForm.vue [hunk 4]
```diff
@@ -164,7 +190,8 @@ export default {
       networkOfferings: [],
       networkOfferingLoading: false,
       selectedNetworkOffering: {},
-      accountVisible: this.isAdminOrDomainAdmin()
+      accountVisible: this.isAdminOrDomainAdmin(),
```
Comment (rhtyd) [inline src/views/network/CreateL2NetworkForm.vue:193]: @davidjumani did you check against legacy UI if we should show pvlan options to domain admins?

**Pull request merged**